packages/types/src/metadata/decorate/storage/index.spec.ts

```diff
@@ -36,6 +36,10 @@ describe('decorateStorage', (): void => {
     expect(() => query.timestamp.now()).not.toThrow();
   });
 
+  it('should return a value if the storage function has the correct arguments', (): void => {
+    expect(() => query.staking.erasStakers(1, keyring.alice.address)).not.toThrow();
```

Comment:
Shouldn't it also verify that calling the key function is working properly ?

Reply:
This is on the storage decorator (named `creator` in the API), the responsibility of the method is to actually return the key. (This is slightly different than in the API itself, here it returns a prefixed `Uint8Array`, hence the slight hoops in the API to convert to hex key)

Reply:
Thank you for the explanations and the quick fix